ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BlockProcessorUtil.java

```diff
@@ -80,8 +82,35 @@ public static void verify_and_update_randao(BeaconState state, BeaconBlock block
     // state.latest_randao_mixes[get_current_epoch(state) % LATEST_RANDAO_MIXES_LENGTH] =
     // xor(get_randao_mix(state, get_current_epoch(state)), hash(block.randao_reveal))
     int index = toIntExact(epoch) % Constants.LATEST_RANDAO_MIXES_LENGTH;
-    state.getLatest_randao_mixes().get(index).xor(Hash.keccak256(epochBytes));
+    Bytes32 latest_randao_mixes = state.getLatest_randao_mixes().get(index);
+    state.getLatest_randao_mixes().set(index, latest_randao_mixes.xor(Hash.keccak256(epochBytes)));
   }
 
-  public static void tally_pow_receipt_root_vote(BeaconState state, BeaconBlock block) {}
+  /**
+   * https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#eth1-data
+   *
+   * @param state
+   * @param block
+   */
+  public static void tally_pow_receipt_root_vote(BeaconState state, BeaconBlock block) {
+    /*
+     Eth1 data
+     If block.eth1_data equals eth1_data_vote.eth1_data for some eth1_data_vote
+       in state.eth1_data_votes, set eth1_data_vote.vote_count += 1.
+     Otherwise, append to state.eth1_data_votes
+       a new Eth1DataVote(eth1_data=block.eth1_data, vote_count=1).
+    */
+
+    int index = 0;
+    for (Eth1DataVote vote : state.getEth1_data_votes()) {
+      if (block.getEth1Data().equals(vote.getEth1_data())) {
+        UnsignedLong voteCount = vote.getVote_count().plus(UnsignedLong.ONE);
+        vote.setVote_count(voteCount);
+        break;
+      }
+      index++;
+    }
+    List<Eth1DataVote> votes = state.getEth1_data_votes();
```

Comment:
This logic is not in compliance with the description in the comments. According to the commented description, this should only happen in the "otherwise" case, i.e. and else condition for the above if on Line 106.

Reply:
yep, should have labeled it as WIP